pkg/skaffold/schema/versions.go

```diff
@@ -301,3 +284,93 @@ func parseConfig(buf []byte, factories []func() util.VersionedConfig) ([]util.Ve
 	}
 	return cfgs, nil
 }
+
+// checkCompatibility makes sure the schema versions in SchemaVersionsV1 and SchemaVersionsV2 are not used together.
```

Comment:
it's not very clear why the corresponding `latest.Version` is being returned here. Would be good to explain here

Reply:
Yeah, renamed to getLatestFromCompatibilityCheck.